lib/AST/ASTImporter.cpp

```diff
@@ -2417,21 +2423,33 @@ Decl *ASTNodeImporter::VisitFunctionDecl(FunctionDecl *D) {
   QualType T = Importer.Import(FromTy);
   if (T.isNull())
     return nullptr;
+  AlreadyImportedD = Importer.GetAlreadyImportedOrNull(D);
+  if (AlreadyImportedD)
+    // the one verified case for this check: this case can be true and without
+    // it a duplicate ToFunction would be generated
+    return AlreadyImportedD;
 
   // Import the function parameters.
   SmallVector<ParmVarDecl *, 8> Parameters;
   for (auto P : D->parameters()) {
     ParmVarDecl *ToP = cast_or_null<ParmVarDecl>(Importer.Import(P));
     if (!ToP)
       return nullptr;
+    AlreadyImportedD = Importer.GetAlreadyImportedOrNull(D);
```

Comment:
Why do we need these three lines? Didn't we check already the same thing in line 2398-2400?